bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java

```diff
@@ -205,13 +205,10 @@ private void scan() {
                 logger.trace("Checking candidate: {}", candidate.getUID());
 
                 // skip scanning if already installed
-                AddonService tmpAddonService = addonService;
-                if (tmpAddonService != null) {
-                    Addon a = tmpAddonService.getAddon(candidate.getUID(), null);
-                    if (a != null && a.isInstalled()) {
-                        logger.trace("Skipping {}, already installed", candidate.getUID());
-                        continue;
-                    }
+                Addon addon = addonService.getAddon(candidate.getUID(), null);
```

Comment:
There are several `AddonService` implementations so you may need to check them all like the `AddonResource` does.

Reply:
I'll have a look. Is it really that complicated to find out if an addon is installed?

Reply:
Yes because there has not been anything else that needed all the logic that is in the `AddonResource` so far.

Reply:
@wborn Thanks for pointing me to the right place. I reworked my PR, could you have a look please?